ruby/tests/basic.rb

```diff
@@ -254,6 +254,15 @@ def test_map_encode_decode
                     "b" => TestMessage2.new(:foo => 2)}
     end
 
+    def test_protobuf_decode_json_ignore_unknown_fields
+      m = TestMessage.decode_json({ optional_string: "foo", not_in_message: "some_value" }.to_json, { ignore_unknown_fields: true })
+      assert_equal m.optional_string, "foo"
+      e = assert_raise Google::Protobuf::ParseError do
```

Comment:
If ignore_unknown_fields is true, why is there still a parser error?
BTW, could you add tests for both ignore_unknown_fields is true and false?

Reply:
I'm not passing in the ignore_unknown_fields flag in this second assert. So this test covers both cases. Would it be better if that was split across two different tests?